pkgs/os-specific/linux/firmware/fwupd/default.nix

```diff
@@ -265,6 +271,14 @@ let
       sed -i 's/test(.*)//' libfwupdplugin/meson.build
       # in nixos test tries to chmod 0777 $out/share/installed-tests/fwupd/tests/redfish.conf
       sed -i "s/get_option('tests')/false/" plugins/redfish/meson.build
+
+      # Device tests use device emulation and need to download emulation data from
+      # the internet, which does not work on our test VMs.
+      # It's probably better to disable these tests for NixOS by setting
+      # the device-tests directory to /dev/null.
+      # For more info on device emulation, see:
+      #   https://github.com/fwupd/fwupd/blob/eeeac4e9ba8a6513428b456a551bffd95d533e50/docs/device-emulation.md
+      sed -i "s:join_paths(datadir, 'fwupd', 'device-tests'):'/dev/null':" data/installed-tests/meson.build
```

Comment:
@jtojnar device tests are disabled here, by replacing the directory containing the test by `/dev/null`.

Reply:
Please use `substituteInPlace` unless a regular expression is needed. Unlike `sed`, that will notify us when the replacement no longer works.